hail/src/main/scala/is/hail/expr/ir/EmitStream.scala

```diff
@@ -784,6 +784,32 @@ object EmitStream {
                 k(TypedTriplet(a, bodyt)))
             }).map(_.untyped)
 
+        case RunAggScan(array, name, init, seqs, result, signature) =>
+          val (newContainer, aggSetup, aggCleanup) = AggContainer.fromFunctionBuilder(signature.toArray, fb, "array_agg_scan")
+
+          val producerElementPType = coerce[PStreamable](array.pType).elementType
+          val resultPType = result.pType
+          implicit val eP = TypedTriplet.pack(producerElementPType)
+          implicit val aP = TypedTriplet.pack(resultPType)
+          val (eltm, eltv) = eP.newFields(fb, "aggscan_elt")
+          val (postm, postv) = aP.newFields(fb, "aggscan_new_elt")
+          val bodyEnv = env.bind(name -> ((typeToTypeInfo(producerElementPType), eltm, eltv)))
+          val cInit = emitter.emit(init, env, er, Some(newContainer))
+          val seqPerElt = emitter.emit(seqs, bodyEnv, er, Some(newContainer))
+          val postt = emitter.emit(result, bodyEnv, er, Some(newContainer))
+
+          emitStream(array, env)
+            .contMap[EmitTriplet] { (eltt, k) =>
+              Code(
+                TypedTriplet(producerElementPType, eltt).storeTo(eltm, eltv),
+                TypedTriplet(resultPType, postt).storeTo(postm, postv),
```

Comment:
It doesn't look like the `postm`, `postv` fields are necessary. Couldn't the returned emit triplet just forward the `m` and `v` from `postt`?

Reply:
doesn't work, generates invalid bytecode. Don't want to dig into that now.

Reply:
Huh, okay.